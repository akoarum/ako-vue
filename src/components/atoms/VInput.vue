<template>
  <input v-model="model" :style="styled" class="vInput" />
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  props: {
    value: { type: String, required: true },
    width: { type: String }
  },
  computed: {
    model: {
      get(): string {
        return this.value
      },
      set(newValue: string) {
        this.$emit('input', newValue)
      }
    },
    styled(): { [key: string]: string } {
      const styled: { [key: string]: string } = {}

      if (this.width) {
        styled['--width'] = this.width
      }

      return styled
    }
  }
})
</script>

<style lang="scss" scoped>
.vInput {
  --width: 100%;
  box-sizing: border-box;
  border: 1px solid #ccc;
  font-size: 1.6rem;
  line-height: 1.2;

  @include media-min {
    padding: 8px 10px;
    width: var(--width);
  }

  @include media-max {
    padding: 7px 10px;
    width: 100%;
  }
}

[aria-invalid='true'] {
  border-color: $color-caution;
  background: #fceaf0;
}
</style>
