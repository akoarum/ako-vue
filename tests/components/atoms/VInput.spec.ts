import { shallowMount } from '@vue/test-utils'
import VInput from '~/components/atoms/VInput.vue'

describe('VInput', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VInput, {
      propsData: {
        value: ''
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styled: propsを基にスタイルを返す', () => {
      expect(wrapper.vm.styled).toEqual({})

      wrapper.setProps({ width: '280px' })
      expect(wrapper.vm.styled).toEqual({ '--width': '280px' })
    })
  })
})
