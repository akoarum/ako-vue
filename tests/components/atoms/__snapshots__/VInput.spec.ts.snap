// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VInput initialized snapshot 1`] = `
<input
  class="vInput"
/>
`;
